### ISSUE TITLE
Abnormal results during DPO training

### ISSUE THREAD

**Issue by wjjwyj** [huggingface/trl#4023]
I tried different parameters, different datasets, and the result is always 0.6934 for loss, and always 0 for grad

The training process is as follows：
**{'loss': 0.6934, 'grad_norm': 0.0, 'learning_rate': 4.4999999999999996e-05, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': -20.380931854248047, 'logps/rejected': -55.611717224121094, 'logits/chosen': 0.04637949541211128, 'logits/rejected': 0.6950775384902954, 'epoch': 0.0}
{'loss': 0.6934, 'grad_norm': 0.0, 'learning_rate': 9.5e-05, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': -19.491588592529297, 'logps/rejected': -55.84765625, 'logits/chosen': 0.028241563588380814, 'logits/rejected': 0.7012389898300171, 'epoch': 0.0}
{'loss': 0.6934, 'grad_norm': 0.0, 'learning_rate': 0.000145, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': -20.011266708374023, 'logps/rejected': -56.07890701293945, 'logits/chosen': 0.033783912658691406, 'logits/rejected': 0.681854248046875, 'epoch': 0.0}
{'loss': 0.6934, 'grad_norm': 0.0, 'learning_rate': 0.00019500000000000002, 'rewards/chosen': 0.0, 'rewards/rejected': 0.0, 'rewards/accuracies': 0.0, 'rewards/margins': 0.0, 'logps/chosen': -20.203527450561523, 'logps/rejected': -57.808204650878906, 'logits/chosen': 0.04809093475341797, 'logits/rejected': 0.7075744867324829, 'epoch': 0.0}**
.
The adapter I'm using is a fine-tuned adapter from the SFT stage.. The datasets used were code and math.
My code is attached, hopefully some kind people can tell me the problem, it's been bugging me for 4 days [now.❤😖]

[m_dpo_llama3.py](https://github.com/user-attachments/files/22188671/m_dpo_llama3.py)

**Comment by qgallouedec** [comment]
Hi, please share the system info

**Comment by wjjwyj** [comment]
> Hi, please share the system info您好，请分享系统信息

I am using **one RTX 4090** with DeepSpeed acceleration and 4-bit quantization.
I also attempted training with two RTX 4090s, but the results remained unchanged.
My image configuration includes **PyTorch 2.1.0, Python 3.10 (Ubuntu 22.04), and CUDA 12.1.**
I am using the source code library of trl.

**Comment by drumzhang** [comment]
hi, I have the same question. Have you solved?

**Comment by wjjwyj** [comment]
> hi, I have the same question. Have you solved?

yes, is your code similar to mine? My error occurred because I used LORA during the sft phase and again during the dpo phase, causing issues with my adapter. I've forgotten the exact command now, but Hugging Face has one that can print out your DPO adapter structure. Check if your adapter changed to “none” at any step.
